plugins/outputs/mqtt/mqtt.go

```diff
@@ -58,6 +61,16 @@ func (*MQTT) SampleConfig() string {
 	return sampleConfig
 }
 
+func (m *MQTT) Init() error {
+	m.template = template.Must(template.New("topic_name").Parse(m.Topic))
+	for _, p := range strings.Split(m.Topic, "/") {
+		if strings.ContainsAny(p, "#+") {
```

Comment:
I don't know if this is possible to handle as part of the go-template. If it's possible, I'd appreciate some further guidance.

Reply:
No. I think keeping it as you have it is the right thing to do. Please also check for an empty string here!

Reply:
sure

Reply:
actually, this was intentionally done as is. If you see this: https://github.com/influxdata/telegraf/pull/12224/files#diff-812adb9d8be29ed391deb32c15e3a8010d7bed23cc79cfc0b1b4de9496df0e74R43

An empty string would use the plugin name. This is the current behaviour. If you think I need to enforce a change here, I am happy to follow direction, but I would to clarify first. TMK, this could be considered as a backward incompatible change or breaking change of previous implicit behaviour. However, we may argue that was an undocumented behaviour and that this is fixing a bug.

Your call how you think we should proceed though, happy to enforce stricter rules.